pages/profile/[username].tsx

```diff
@@ -86,9 +87,4 @@ const UserProfile: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
   )
 }
 
-export default withQueryLoader<AppData>(
-  {
-    query: GET_APP
-  },
-  UserProfile
-)
+export default UserProfile
```

Comment:
You could also use the custom `with` HOC from the generator, instead of the `use` Query one.

Reply:
My rule of thumb is if the component depends on a single data set, and that data set is tightly integrated with the component, go with HOC, otherwise use hooks

Reply:
Reference:
https://github.com/garageScript/c0d3-app/blob/4c3f4c0c6d89010fe708021d24f66b5c507f2df4/pages/index.tsx#L28

Reply:
> My rule of thumb is if the component depends on a single data set, and that data set is tightly integrated with the component, go with HOC, otherwise use hooks

I have multiple components being powered through one query on the user profile page, so you recommend i should use hooks?

Reply:
Don't look at how the components interact with one another, that only concerns the usage of `props` and `contexts`, in this case, we are only concerned with a single component `UserProfile`. The question is, what data does this component depend on, and how tightly coupled is the component with that data. In the case `UserProfile` depends on the single dataset `userInfo` aka `USER_INFO` right, and that data is very tightly coupled with the component itself, it wouldn't make sense to have a UserProfile without userInfo data. So, I would use an HOC

Reply:
I just realized you will have to use a hook for this case in particular because you need to pass a variable to the query, but the variable will not be known until the component is being rendered, so... you will have to use hooks.

Reply:
References:

https://github.com/garageScript/c0d3-app/blob/4c3f4c0c6d89010fe708021d24f66b5c507f2df4/components/AppNav.tsx#L4
https://github.com/garageScript/c0d3-app/blob/4c3f4c0c6d89010fe708021d24f66b5c507f2df4/components/AppNav.tsx#L54

```
const [action, { data }] = useTypedMutation({
  variables: {
    ...
  }
})
```

Reply:
Actually, I just had another thought, you might be able to double wrap with 2 HOCs to make it work, a `withRouter` HOC from next.js to provide the variable you need, and then the `withTypedQuery` HOC from graphql/index.tsx. Up to you though, these are only suggestions :)